tests/rbd_mirror/test_rbd_mirror_snapshot_metrics.py

```diff
@@ -0,0 +1,247 @@
+    1. Two Clusters must be up and running to create pool
+    2. We need atleast one client node with ceph-common package,
+       conf and keyring files on each node.
+
+    Test Case Flow:
+        1. Create a pool on both clusters.
+        2. Create an Image on primary mirror cluster in same pool.
+        3. Configure mirroring (peer bootstrap) between two clusters.
+        4. Enable image mode snapshot based mirroring on the pool respectively.
+        5. Start running IOs on the primary image.
+        6. verify snapshot mirror based performance conter metrics.
+"""
+import json
+import re
+
+from tests.rbd_mirror import rbd_mirror_utils as rbdmirror
+from utility.log import Log
+
+log = Log(__name__)
+
+
+def validate_snap_metrics(mirror_node):
+    """Method to validate snapshot based mirroring metrics.
+    Args:
+        mirror_node: rbd-mirror daemon node.
+
+    Example:
+        "rbd_mirror_snapshot": {
+            "labels": {},
+            "counters": {
+                "snapshots": 13404,
+                "sync_time": {
+                    "avgcount": 13404,
+                    "sum": 42.406661993,
+                    "avgtime": 0.003163731
+                },
+                "sync_bytes": 1693450240
+            }
+
+        "rbd_mirror_snapshot_image": {
+            "labels": {
+                "image": "io_set_1",
+                "namespace": "",
+                "pool": "test_pool"
+            },
+            "counters": {
+                "snapshots": 49,
+                "sync_time": {
+                    "avgcount": 49,
+                    "sum": 6.935865147,
+                    "avgtime": 0.141548268
+                },
+                "sync_bytes": 1588592640,
+                "remote_timestamp": 1682935620.226228164,
+                "local_timestamp": 1682935620.226228164,
+                "last_sync_time": 0.004110399,
+                "last_sync_bytes": 0
+            }
+        }
+    Returns:
+        0 - if test case pass
+        1 - if test case fails
+    """
+    # Set the ceph directory path
+    ceph_dir_path = "/var/run/ceph"
+
+    try:
+        # install ceph-common required packages
+        mirror_node.exec_command(
+            sudo=True,
+            cmd="yum install -y ceph-common --nogpgcheck",
+            check_ec=False,
+        )
+
+        # List the contents of the ceph directory
+        directory_contents = mirror_node.exec_command(
+            cmd=f"ls {ceph_dir_path}", sudo=True
+        )
+
+        # Check if there are any directories in the current directory
+        if directory_contents:
+            target_directory = directory_contents[0].strip()
+
+            # List the contents of the target directory
+            directory_contents = mirror_node.exec_command(
+                cmd=f"ls {ceph_dir_path}/{target_directory}/", sudo=True
+            )
+
+            # Find the file with the specific pattern
+            file_pattern = "ceph-client.rbd-mirror.*"
```

Comment:
I think we can create a common method for creating symbolic link and fetching the metrics, because these lines of code from line number 70 to line number 129 look same in both journal and snapshot based metrics files. 

Please create a common method for the same in ceph/rbd/workflows/rbd_mirror_metrics.py and import that and use it in these test modules.

Reply:
Thanks @manasagowri for the suggestion done the changes accordingly,
please review it
success logs for the latest code changes
http://magna002.ceph.redhat.com/cephci-jenkins/cephci-run-UBJTJ3